streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -610,17 +610,32 @@ public void setStreamsUncaughtExceptionHandler(final java.util.function.Consumer
         this.streamsUncaughtExceptionHandler = streamsUncaughtExceptionHandler;
     }
 
-    public void waitOnThreadState(final StreamThread.State targetState) {
+    public boolean waitOnThreadState(final StreamThread.State targetState, long timeoutMs) {
+        if (timeoutMs < 0) {
```

Comment:
for the non timeout uses

Reply:
I think if you fix the semantics of `removeStreamThread()` to match that of `close()` then there's no need for a `-1` sentinel, in which case we should just throw an `IllegalArgumentException` here (or it's probably better to check and throw that in the actual `removeStreamThread(timeout)` call to fail fast